cider/function/SubstraitFunctionCiderMappings.h

```diff
@@ -0,0 +1,153 @@
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef CIDER_FUNCTION_SUBSTRAITFUNCTIONMAPPINGS_H
+#define CIDER_FUNCTION_SUBSTRAITFUNCTIONMAPPINGS_H
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include "type/plan/Analyzer.h"
+#include "util/sqldefs.h"
+
+using FunctionSQLScalarOpsMappings = std::unordered_map<std::string, SQLOps>;
+using FunctionSQLAggOpsMappings = std::unordered_map<std::string, SQLAgg>;
+using FunctionSQLOpSupportTypeMappings =
+    std::unordered_map<std::string, OpSupportExprType>;
+
+class SubstraitFunctionCiderMappings {
+ public:
+  const SQLOpsPtr getFunctionScalarOp(const std::string& function_name) const {
+    const auto& scalar_op_map = scalarMappings();
+    auto iter = scalar_op_map.find(function_name);
+    if (iter == scalar_op_map.end()) {
+      return nullptr;
+    }
+    return std::make_shared<SQLOps>(iter->second);
+  }
+
+  const SQLAggPtr getFunctionAggOp(const std::string& function_name) const {
+    const auto& agg_op_map = aggregateMappings();
+    auto iter = agg_op_map.find(function_name);
+    if (iter == agg_op_map.end()) {
+      return nullptr;
+    }
+    return std::make_shared<SQLAgg>(iter->second);
+  }
+
+  const OpSupportExprTypePtr getFunctionOpSupportType(
+      const std::string& function_name) const {
+    const auto& op_support_map = opsSupportTypeMappings();
+    auto iter = op_support_map.find(function_name);
+    if (iter == op_support_map.end()) {
+      return nullptr;
+    }
+    return std::make_shared<OpSupportExprType>(iter->second);
+  }
+
+  /// scalar function names in difference between engine own and Substrait.
+  virtual const FunctionSQLScalarOpsMappings scalarMappings() const {
+    static const FunctionSQLScalarOpsMappings scalarMappings{
+        {"lt", SQLOps::kLT},
+        {"and", SQLOps::kAND},
+        {"or", SQLOps::kOR},
+        {"not", SQLOps::kNOT},
+        {"gt", SQLOps::kGT},
+        {"eq", SQLOps::kGT},
+        {"equal", SQLOps::kEQ},
+        {"neq", SQLOps::kEQ},
+        {"ne", SQLOps::kEQ},
+        {"not_equal", SQLOps::kNE},
+        {"gte", SQLOps::kGE},
+        {"lte", SQLOps::kLE},
+        {"multiply", SQLOps::kMULTIPLY},
```

Comment:
For the key, it should also include input types, such as gt:int_int, follow the function lookup rule in Chaojun's code.

Reply:
First will use velox's substrait function lookup to verify the function based on function name and function argument type and function return type. If verify success then will use function name to look up the SQLOps info.